pkg/fanal/image/image.go

```diff
@@ -95,14 +95,21 @@ func NewContainerImage(ctx context.Context, imageName string, option types.Docke
 		errs = multierror.Append(errs, err)
 	}
 
+	if len(option.Credentials) == 0 {
+		option.Credentials = append(option.Credentials, types.Credential{}) // no credential use-case
```

Comment:
@knqyf263 @chen-keinan what was point of this row? By using this row, you will remove backwards compatibility AND when you are not removing existing UserName and Password structs, it means that people cannot detect that something has modified. It took 3 days for us to find out what is going on because all the auths were failing.

Instead of using empty value here, the correct way is

```
		option.Credentials = append(option.Credentials, types.Credential{
			UserName: option.UserName,
			Password: option.Password,
		})
```

Then the solution is backwards compatible.

Reply:
@zetaab could you please advice what is the use case which is not backward supported ? 
if `user / password` env variable are not empty then it should not enter `L99`